backend/src/utils/route-security.ts

```diff
@@ -257,24 +257,36 @@ export const sanitizeNotebookForSecurity = async <
 
   // Container based items
   secureNotebook?.spec?.template?.spec?.containers?.forEach((container) => {
-    // Secrets & ConfigMaps
+    // Configmaps
     container.env?.forEach((env) => {
-      if (env.name.startsWith('jupyterhub-singleuser-profile-')) {
-        // Env var for a configmap or secret we generated
-        const allowedValue = `jupyterhub-singleuser-profile-${translatedUsername}-env`;
-        if (env.name !== allowedValue) {
+      if (env.valueFrom?.configMapKeyRef?.name?.startsWith('jupyterhub-singleuser-profile-')) {
```

Comment:
I don't think we need to check the `.startsWith`.  I think the only time we allow `secretKeyRef` or `configMapKeyRef` we should set the name to allowed name.  We don't want them injecting:
```
          - name: malicious
            valueFrom:
              secretKeyRef:
                key: malicious
                name:some-other-secret-stolen
```
which I think would be skipped by this check

Reply:
Yes, that's a great idea, I'm gonna restrict it to that.